boot/assets.go

```diff
@@ -188,6 +198,42 @@ func (t *trackedAsset) equal(other *trackedAsset) bool {
 		t.hash == other.hash
 }
 
+func isAlreadyTracked(current []*trackedAsset, newAsset *trackedAsset) bool {
+	for _, ta := range current {
+		if newAsset.equal(ta) {
+			return true
+		}
+	}
+	return false
+}
+
+func isAlreadyTrackedInBootMap(bam bootAssetsMap, newAsset *trackedAsset) bool {
+	return isAssetHashTrackedInBootMap(bam, newAsset.name, newAsset.hash)
+}
+
+func isAssetHashTrackedInBootMap(bam bootAssetsMap, assetName, assetHash string) bool {
+	if bam == nil {
+		return false
+	}
+	hashes, ok := bam[assetName]
+	if !ok {
+		return false
+	}
+	return strutil.ListContains(hashes, assetHash)
+}
+
+func trackedAssetsToBootMap(trackedAssets []*trackedAsset) bootAssetsMap {
```

Comment:
is this used ?

Reply:
Hmm no, it's part of #9160 but found it's way here when I was carving out the patches from a larger branch.